test/themes/base.theme.git.bats

```diff
@@ -0,0 +1,390 @@
+#!/usr/bin/env bats
+
+load ../test_helper
+load ../../lib/composure
+
+cite _about _param _example _group _author _version
+
+load ../../lib/helpers
+load ../../themes/githelpers.theme
+load ../../themes/base.theme
+
+add_commit() {
+  local file_name="general-${RANDOM}"
+  touch "${file_name}"
+  echo "" >> "${file_name}"
+  git add "${file_name}"
+  git commit -m"message"
+}
+
+setup_repo_with_upstream() {
+  upstream="$(mktemp -d)"
+  pushd "$upstream" > /dev/null
+  git init . > /dev/null
+  add_commit > /dev/null
+  git checkout -b branch-two
+  git checkout -b gone-branch
+  git checkout master
+  popd > /dev/null
+
+  downstream="$(mktemp -d)"
+  pushd "$downstream" > /dev/null
+  git init . > /dev/null
+  add_commit > /dev/null
+  git remote add my-remote "$upstream"
+  git fetch my-remote
+  git branch -u my-remote/master > /dev/null
+  popd > /dev/null
+
+  pushd "$upstream" > /dev/null
+  git branch -d gone-branch > /dev/null
+  popd > /dev/null
+
+  pushd "$downstream" > /dev/null
+  git fetch my-remote
+  popd > /dev/null
+
+  echo "$downstream"
+}
+
+@test 'themes base: SCM_BRANCH: when tracking a remote branch: it shows the commits ahead and behind' {
+  pre="\$(_git-friendly-ref)"
+
+  remote="$(mktemp -d)"
+  pushd "$remote"
+  git init .
+  add_commit
+  add_commit
+  popd
+
+  clone="$(mktemp -d)"
+  pushd "$clone"
+  git clone "$remote" clone
+  cd clone
+
+  git_prompt_vars
+  assert_equal "$SCM_BRANCH" "${pre}"
+
+  add_commit
+
+  git_prompt_vars
+  assert_equal "$SCM_BRANCH" "${pre} ↑1"
+
+  add_commit
+
+  git_prompt_vars
+  assert_equal "$SCM_BRANCH" "${pre} ↑2"
+  popd
+
+  pushd "$remote"
+  add_commit
+  add_commit
+  add_commit
+  popd
+
+  pushd "$clone/clone"
+  git fetch
+
+  git_prompt_vars
+  assert_equal "$SCM_BRANCH" "${pre} ↑2 ↓3"
+
+  git reset HEAD~2 --hard
+
+  SCM_GIT_BEHIND_CHAR="↓"
+  git_prompt_vars
+  assert_equal "$SCM_BRANCH" "${pre} ↓3"
+
+  popd
+}
+
+@test 'themes base: SCM_BRANCH: when stashes exist: it shows the number of stashes' {
+  pre="\$(_git-friendly-ref)"
+
+  clone="$(mktemp -d)"
```

Comment:
These four lines (`mktemp, pushd, git init, add_commit`) are repeated several times in the test cases. They look like an opportunity for refactoring into a setup function...

Reply:
Makes sense. I'll refactor that.